src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAOSpec.scala

```diff
@@ -23,60 +27,110 @@ import scala.concurrent.Future
-    bindingFuture.flatMap(_.unbind())
+    bindingFutureHealthy.flatMap(_.unbind())
+    bindingFutureUnhealthy.flatMap(_.unbind())
     super.afterAll()
   }
 
-  val expectedStatus = SubsystemStatus(true, None)
+  val returnedOkStatus = StatusCheckResponse(true, Map(OpenDJ -> SubsystemStatus(true, None), GoogleIam -> SubsystemStatus(true, None)))
+  val expectedOkStatus = StatusCheckResponse(true, Map.empty)
+  val notOkStatus = StatusCheckResponse(false, Map(OpenDJ -> SubsystemStatus(false, Option(List("OpenDJ is down. Panic!"))), GoogleIam -> SubsystemStatus(true, None)))
+
   val expectedPet = WorkbenchEmail("pet-1234567890@test-project.iam.gserviceaccount.com")
-  val defaultUserInfo = UserInfo(OAuth2BearerToken("accessToken"), WorkbenchUserId("user1"), WorkbenchEmail("user1@example.com"), 0)
-  val otherUserInfo = UserInfo(OAuth2BearerToken("tokenFoo"), WorkbenchUserId("user2"), WorkbenchEmail("user2@example.com"), 0)
+  val defaultUserInfo = UserInfo(OAuth2BearerToken("accessToken1"), WorkbenchUserId("user1"), WorkbenchEmail("user1@example.com"), 0)
+  val errorUserInfo = UserInfo(OAuth2BearerToken("accessToken2"), WorkbenchUserId("user2"), WorkbenchEmail("user2@example.com"), 0)
+  val unknownUserInfo = UserInfo(OAuth2BearerToken("accessToken3"), WorkbenchUserId("user3"), WorkbenchEmail("user3@example.com"), 0)
+  val googleProject = GoogleProject("my-google-proj")
+
+  def getPet(auth: Option[Authorization]): ToResponseMarshallable = {
+    auth match {
+      case Some(Authorization(credentials)) if credentials.token() == defaultUserInfo.accessToken.token => expectedPet
+      case Some(Authorization(credentials)) if credentials.token() == errorUserInfo.accessToken.token => StatusCodes.InternalServerError -> ErrorReport(new Exception("my message"))
+      case _ => StatusCodes.Unauthorized
+    }
+  }
+
+  val unhealthyRoute: Route =
+    path("status") {
+      get {
+        complete(StatusCodes.InternalServerError -> notOkStatus)
+      }
+    }
 
   val backendRoute: Route =
     path("status") {
       get {
-        complete(expectedStatus)
+        complete(returnedOkStatus)
       }
     } ~
+    // deprecated!
     pathPrefix("api") {
       path("user" / "petServiceAccount") {
         get {
           extractRequest { request =>
-            request.header[Authorization] match {
-              case Some(Authorization(credentials)) if credentials.token() == defaultUserInfo.accessToken.token =>
-                complete(expectedPet)
-              case _ =>
-                complete(StatusCodes.BadRequest)
-            }
+            complete(getPet(request.header[Authorization]))
+          }
+        }
+      } ~
+      path("google" / "user" / "petServiceAccount" / Segment) { googleProject =>
+        get {
+          extractRequest { request =>
+            complete(getPet(request.header[Authorization]))
           }
         }
       }
     }
 
   val dao = new HttpSamDAO("http://localhost:9090")
```

Comment:
Please use vals for the port numbers here and above, to make it explicit

Reply:
👍